src/hipchatbot.js

```diff
@@ -0,0 +1,79 @@
+import Hipchatter from 'hipchatter';
+import trebek from './trebek';
+import winston from 'winston';
+import App from './models/App';
+import { HIPCHAT_CALLBACK_URL } from './config';
+
+export default class HipchatBot {
+  constructor(express) {
+    this.express = express;
+    this.start();
+  }
+
+  async start() {
+    this.app = await App.get();
+    this.hipchatter = new Hipchatter(this.app.apiToken);
+    this.registerWebhook();
+    this.express.post('/webhook', this.onMessage.bind(this));
+  }
+
+  onMessage(req) {
+    const text = req.body.item.message.message;
+    const subtype = req.body.event;
+    const channel_name = req.body.item.room.name;
+    const channel_id = (req.body.item.room.id).toString();
+    const timestamp = Date.parse(req.body.item.message.date) / 1000;
+    const user_id = req.body.item.message.from.id;
+    const user_name = req.body.item.message.from.mention_name;
```

Comment:
Fun-fact: You can replace all of these lines with this:

```javascript
const { event: subtype, item: { room: { name: channel_name, id: channel_id }, message: { date: timestamp, message: text, from: { id: user_id, mention_name: user_name } } } } = req.body;
```

But you probably shouldn't.